src/providers/code-actions/commonActions.ts

```diff
@@ -3,23 +3,27 @@ import {DiagnosticCode} from '../../services/common';
 import {VscodeCommands} from '../../utils/commands';
 import {CommandParameters} from '../../cli-wrapper/constants';
 import {IgnoreCommandConfig} from '../../types/commands';
+import {scanResultsService} from '../../services/ScanResultsService';
 
 export const createIgnoreRuleAction = (
     diagnostics: vscode.Diagnostic[], diagnosticCode: DiagnosticCode, document: vscode.TextDocument
 ): vscode.CodeAction => {
+  const detection = scanResultsService.getDetectionById(diagnosticCode.uniqueDetectionId);
+  const ruleId = detection?.detection_rule_id;
```

Comment:
Do we want to handle the case where `detection` is not present? Seems strange to be able to perform an action on a null detection / rule ID, but perhaps I am missing context here.

Reply:
imo this is not possible. because the action creator are triggered by diagnostics. with the provided diagnostics code. when the diagnostic is created it means we will have detection in the scan results service. if it's not found it's a bug. 

root of the bug:
1. someone not saved it to "DB" -- they must save it
2. someone deleted it from "DB" -- but they must delete diagnostics at the same time

for now, we don't delete items, only override with new results

ofc we can handle this case but it will be harder to find one of the mentioned bugs in the future. also, it's handled already by `?`? it will not crash the extension but will show it without message, for example

Reply:
^ Thanks for explaining; makes sense now!